ws-server/src/server.rs

```diff
@@ -252,9 +257,31 @@ async fn background_task(
 		while !stop_server2.shutdown_requested() {
 			match rx.next().await {
 				Some(response) => {
-					tracing::debug!("send: {}", response);
-					let _ = sender.send_text(response).await;
-					let _ = sender.flush().await;
+					if response.len() > max_request_body_size as usize {
+						tracing::warn!(
+							"WS Transport error: response to method call too large {}, max: {}",
+							response.len(),
+							max_request_body_size
+						);
+						// TODO(niklasad1): include `id` in the response and send back `response too big; id=id` here?!
+						// also we could terminate the connection...
```

Comment:
/cc @maciejhirsz 

currently we have `UnboundedChannel::<String>` but we would like the extract the `ID` from there in case the response is too big and send `{"jsonrpc":"2.0", "result":"Response to call was too big", "id":<ID>}`

however, we could simply use `Id::Null` or `Error` here too these will likely be thrown by the jsonrpc clients anyway I guess...

Reply:
It sounds like it could be something that's cought earlier in the pipeline, in the module itself perhaps?

We could also do a custom `Write` impl that takes care of the max buffer size [during serialization](https://docs.rs/serde_json/1.0.69/serde_json/fn.to_writer.html).

Reply:
Why not make this into an error, `InvalidRequest` (code `-32600`)?

Reply:
> It sounds like it could be something that's cought earlier in the pipeline, in the module itself perhaps?

> We could also do a custom Write impl that takes care of the max buffer size during serialization.

Good idea

> Why not make this into an error, InvalidRequest (code -32600)?

yeah, an error sounds good I forgot that the error also contains request ID.